dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/apps/AppsResourceTest.java

```diff
@@ -429,13 +459,15 @@ public void Test_Create_App_Descriptor_Then_Create_App_Integration_Then_Delete_A
     }
 
     @Test
-    public void Test_Protected_Hidden_Secret() {
-
-        final Map<String, Param> initialParamsMap = ImmutableMap.of(
-                "param1", Param.newParam("val-1", false, Type.STRING, "label", "hint", true),
-                "param2", Param.newParam("true", false, Type.BOOL, "label", "hint", true),
-                "param3", Param.newParam("val-2", true, Type.STRING, "label", "hint", true),
-                "param4", Param.newParam("true", true, Type.BOOL, "label", "hint", true)
+    public void Test_Protected_Hidden_Secret_And_Values_Returned_Match_Descriptor() {
+
+        final List<String> orderedParamNames = ImmutableList.of("param1","param2","param3","param4");
+        //This is how the descriptor looks like.
+        final SortedMap<String, ParamDescriptor> appDescriptorParamsMap = ImmutableSortedMap.of(
+                orderedParamNames.get(0), ParamDescriptor.newParam("val-1", false, Type.STRING, "label", "hint", true),
+                orderedParamNames.get(1), ParamDescriptor.newParam("true", false, Type.BOOL, "label", "hint", true),
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "true" appears 4 times in this file; the first occurrence is on line 468](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5289044)